src/fparser/two/Fortran2003.py

```diff
@@ -7203,102 +7208,456 @@ def match(string):
-                    Format_Item(p+repmap(r.lstrip()))
+        '''Implements the matching for the C1002 Format Item constraint. The
+        constraints specify certain combinations of format items that
+        do not need a comma to separate them. Rather than sorting this
+        out when parsing the list, it was decided to treat these
+        separately and match them in this class. As a result the
+        generated class hierarchy is a little more complicated.
+
+        :param str string: The string to check for conformance with a \
+                           C1002 format item constraint.
+        :return: `None` if there is no match, otherwise a tuple of \
+        size 2 containing a mixture of Control_Edit_Descriptor and \
+        Format_Item classes depending on what has been matched.
+
+        :rtype: `NoneType` or ( \
+        :py:class:`fparser.two.Control_Edit_Desc`, \
+        :py:class:`fparser.two.Format_Item` ) or \
+        (:py:class:`fparser.two.Format_Item`, \
+        :py:class:`fparser.two.Control_Edit_Desc`) or \
+        (:py:class:`fparser.two.Format_Item`, \
+        :py:class:`fparser.two.Format_Item`)
+
+        '''
+        if not string:
+            return None
+        strip_string = string.strip()
+        if len(strip_string) <= 1:
+            return None
+        if strip_string[0] in ':/':
+            # No comma is required after slash edit descriptor (3) or
+            # after a colon edit descriptor (4)
+            return Control_Edit_Desc(strip_string[0]), \
+                Format_Item(strip_string[1:].lstrip())
+        if strip_string[-1] in ':/':
+            # No comma is required before a slash edit descriptor,
+            # when the optional repeat specification is not present
+            # (2), or before a colon edit descriptor (4). Note, if an
+            # optional repeat specification is present it will be
+            # treated as if it is part of the previous item.
+            return Format_Item(strip_string[:-1].rstrip()), \
+                Control_Edit_Desc(strip_string[-1])
+        index = 0
+        # We may have a P edit descriptor (which requires a number
+        # before the 'P') (1) or a slash edit descriptor with a repeat
+        # specifier (3) so look for the repeat specifier.
+        while index < len(strip_string) and (strip_string[index].isdigit() or
+                                             strip_string[index] == ' '):
+            index += 1
+        if index:
+            # We found a possible repeat specifier (which may contain
+            # white space)
+            result = strip_string[index].upper()
+            if result == '/':
+                # We found a possible slash edit descriptor with a
+                # repeat specifier (3).
+                return Control_Edit_Desc(strip_string[:index+1]), \
+                    Format_Item(strip_string[index+1:].lstrip())
+            elif result == 'P':
```

Comment:
pylint says this could just be an `if result` because of the previous `return`

Reply:
Modified.